src/equal.c

```diff
@@ -137,14 +137,49 @@ static int cpl_equal_scalar(const Rcomplex* x, const Rcomplex* y, bool na_equal)
     return real_equal && imag_equal;
   }
 }
+
+// The minimal number of checks required to determine if we need to translate.
+// - CHAR_ENC_TYPE - Don't translate these cases:
+//   (utf8 + utf8), (latin1 + latin1), (unknown + unknown), (bytes + bytes),
+//   (unknown + bytes)
+// - CHAR_IS_BYTES - Don't translate these cases:
+//   (bytes + utf8), (bytes + latin1)
```

Comment:
(bytes + unknown) ?

Reply:
It's a bit tricky, but that case is captured by `CHR_ENC_TYPE()` already. See right above I listed `(unknown + bytes)`. Maybe the name is a bit misleading but it returns either:
- `0` if unknown / bytes
- `4` if latin1
- `8` if utf8